glyphs-reader/src/font.rs

```diff
@@ -3656,4 +3691,34 @@ mod tests {
             *mapping
         );
     }
+
+    #[test]
+    fn cardinal_rotation_contain_exact_zeros_and_ones() {
+        // When Glyphs 3 components are rotated by a 90, 180 or 270 degree angle,
+        // we want to match fontTools Transform.rotate() and have the sine and
+        // cosine terms rounded exactly to 0.0 or ±1.0 to avoid unnecessary diffs
+        // upon rounding transformed coordinates to integer:
+        // https://github.com/googlefonts/fontc/issues/1218
+        assert_eq!(
+            Affine::new([1.0, 0.0, 0.0, 1.0, 0.0, 0.0]),
+            normalized_rotation(0.0)
+        );
+        assert_eq!(
+            Affine::new([0.0, 1.0, -1.0, 0.0, 0.0, 0.0]),
+            normalized_rotation(90.0)
+        );
+        assert_eq!(
+            Affine::new([-1.0, 0.0, 0.0, -1.0, 0.0, 0.0]),
+            normalized_rotation(180.0)
+        );
+        assert_eq!(
+            Affine::new([0.0, -1.0, 1.0, 0.0, 0.0, 0.0]),
+            normalized_rotation(270.0)
+        );
+        // any other angles' sin and cos != (0, ±1) are passed through unchanged
```

Comment:
This is many tests in one, could we split them or use a parameterized test or make it one assert so it tells us all the problems if there are multiple failures?

Also, we should test values very close to multiples of 90. If we believe epsilon does anything - e.g. we can normalize a value that isn't an integer multiple of 90.0 - we should prove it.